packages/client/utils/reactjiPickerRecent.ts

```diff
@@ -0,0 +1,35 @@
+const DEFAULT_ORDER = {
+  heart: 0,
+  tada: 0,
+  smile: 0,
+  rocket: 0,
+  fire: 0,
+  white_check_mark: 0,
+  confused: 0,
+  cry: 0,
+  x: 0
+}
+
+const RECENT_REACTJI_PICKER = 'recentReactjiPicker'
+const MAX_RECENT_EMOJIS = 9
+
+const getRecent = () => {
+  const localStorageRecent = localStorage.getItem(RECENT_REACTJI_PICKER)
+  return localStorageRecent === null ? DEFAULT_ORDER : JSON.parse(localStorageRecent)
```

Comment:
`localStorageRecent` may not be JSON if the value is editing outside of our app (or if we introduce a bug)
if we don't wrap it in try/catch then the whole app could crash.

In order to guard against edge cases, I'd probably use the pre-made solution. emoji-mart already has this functionality built in (see https://github.com/missive/emoji-mart#storage)

I'd propose just adding the following to the parent component & getting rid of the recent array

```ts
  useEffect(() => {
    const existingRecent = localStorage.getItem('emoji-mart.frequently')
    if (existingRecent) return
    localStorage.setItem(
      'emoji-mart.frequently',
      JSON.stringify({
        heart: 0,
        tada: 0,
        smile: 0,
        rocket: 0,
        fire: 0,
        white_check_mark: 0,
        confused: 0,
        cry: 0,
        x: 0
      })
    )
  }, [])
```

Reply:
Thanks for the feedback. Initially, I didn't use their local storage because I couldn't find a proper way to override their default emoji selection. However using `useEffect` in the parent component makes total sense.